src/main/webapp/scripts/personalised-recommendations.js

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const topRecommendations = [
+  {
+    name: "Pret A Manger", 
+    crowd: 3,
+    price: 1
+  },
+  {
+    name: "Wasabi", 
+    crowd: 3,
+    price: 2
+  },
+  {
+    name: "Franco Manca", 
+    crowd: 2,
+    price: 2
+  },
+  {
+    name: "Taco Bell", 
+    crowd: 4,
+    price: 1
+  },
+  {
+    name: "KFC", 
+    crowd: 4,
+    price: 2
+  }
+];
+
+/* Load top 5 recommendations */
+function loadTopRecommendations() {
+  var i;
+  for (i = 0; i < topRecommendations.length; i++) {
+    const topRecommendation = topRecommendations[i];
+    const recommendationBox = document.createElement("div");
+    recommendationBox.className = "recommendation-box";
+    const rankHTML = "<h4>Rank " + i + "</h4>";
```

Comment:
I'm not sure how best to do this. I think rank might seem confusing to the user? In the UI that I already designed, I included just a "top recommendation" message on the very best match, which I think might be more suitable.

Reply:
Setting the top recomendation layout was quite tricky since we want to align all the texts. I could add back the most recommended box if you want to but not in the recommendation box. I was thinkging that ranking with number would be mmore clear.

Reply:
Yes, but the top recommendation box doesn't have to be in each. As long as it's obvious which is the most suited recommendation, then the rest will appear in order. You could remove the ranking and put the 'most recommended' box at the top of the section (above the recommendations)? It'll take away any confusion over what "Rank 1" etc. means